jobs/submanPipeline.groovy

```diff
@@ -0,0 +1,93 @@
+        commitShaSource: [
+            $class: 'ManuallyEnteredShaSource',
+            sha: sha1,
+        ],
+        contextSource: [
+            $class: 'ManuallyEnteredCommitContextSource',
+            context: 'jenkins-pipeline',
+        ],
+        reposSource: [
+            $class: 'ManuallyEnteredRepositorySource',
+            url: repo,
+        ],
+        statusResultSource: [
+            $class: 'ConditionalStatusResultSource',
+            results: [[
+                $class: 'AnyBuildResult',
+                message: message,
+                state: state,
+            ]]
+        ],
+    ])
+}
+
+node('master') { // FIXME rhsm
+    stage('checkout') {
+        deleteDir()
+        setCommitStatus()
+        checkout([
+            $class: 'GitSCM',
+            branches: [[name: '${sha1}']],
+            userRemoteConfigs: [[
+                refspec: '+refs/pull/*/head:refs/remotes/origin/pr/*',
+                url: 'https://github.com/candlepin/subscription-manager.git',
+            ]],
+        ])
+        checkout([
+            changelog: false,
+            poll: false,
+            scm: [
+                $class: 'GitSCM',
+                branches: [[name: 'origin/master']],
+                extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'candlepin-jobs']],
+                userRemoteConfigs: [[
+                    refspec: '+refs/heads/master:refs/remotes/origin/master',
+                    url: 'https://github.com/candlepin/candlepin-jobs.git',
+                ]],
+            ],
+        ])
+        stash name: 'workspace'
+    }
+}
+
+stage('test') {
+    try {
+        parallel(
+            'nose': {
+                node('master') {// FIXME rhsm
+                    deleteDir()
+                    unstash 'workspace'
+                    sh(readFile('candlepin-jobs/resources/subscription-manager-nose-tests.sh'))
+                }
+            },
+            'stylish': {
+                node('master') {// FIXME rhsm
+                    deleteDir()
+                    unstash 'workspace'
+                    sh(readFile('candlepin-jobs/resources/subscription-manager-stylish-tests.sh'))
+                }
+            },
+            'tito': {
+                node('master') {// FIXME rhsm
+                    deleteDir()
+                    unstash 'workspace'
+                    sh(readFile('candlepin-jobs/resources/subscription-manager-tito-tests.sh'))
+                }
+            },
+        )
+        setCommitStatus(REPO, '${sha1}', 'SUCCESS', 'All tests passed.')
+    } catch(e) {
+        setCommitStatus(REPO, '${sha1}', 'FAILURE', 'One or more tests failed.')
+        emailext(
+            subject: 'subscription-manager build failed', // TODO put more info in subject (PR number), etc.
+            body: '', // TODO fill in body
+            to: 'chainsaw@redhat.com',
+        )
+        // TODO figure out how to do IRC notification
```

Comment:
should be similar to: https://github.com/candlepin/candlepin-jobs/blob/master/src/main/groovy/jobLib/rhsmLib.groovy#L42-L56